synapse/tools/backup.py

```diff
@@ -30,23 +46,27 @@ def backup(srcdir, dstdir, compact=True):
 
         for name in list(dnames):
 
-            # Explicitly skip directory names of 'tmp' to avoid backing up temporary files
-            if name == 'tmp':
+            srcpath = s_common.genpath(root, name)
+
+            if name in skipdirs:
+                logger.info(f'skipping dir:{srcpath}')
                 dnames.remove(name)
                 continue
 
-            srcpath = s_common.genpath(root, name)
             dstpath = s_common.genpath(dstdir, relpath, name)
 
             if name.endswith('.lmdb'):
                 dnames.remove(name)
-                backup_lmdb(srcpath, dstpath)
+                backup_lmdb(srcpath, dstpath, compact)
                 continue
 
             logger.info(f'making dir:{dstpath}')
             s_common.gendir(dstpath)
 
         for name in fnames:
+            if name in skipfns:
```

Comment:
Could we make this a glob match so i could specify "layers/*" as a skip?

Reply:
Updated to do a glob match on the relative dir.  I left the exclusion specifically for directories, but we could of course extend it to match filenames as well (either as separate skip option, or just make one generic skip option).